clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3268,12 +3268,29 @@ static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &AL) {
 
   ASTContext &Ctx = S.getASTContext();
 
+  // The arguments to reqd_work_group_size are ordered based on which index
+  // increments the fastest. In OpenCL, the first argument is the index that
+  // increments the fastest, and in SYCL, the last argument is the index that
+  // increments the fastest.
+  //
+  // [[sycl::reqd_work_group_size]] and [[cl::reqd_work_group_size]] are
+  // available in SYCL modes and follow the SYCL rules.
+  // __attribute__((reqd_work_group_size)) is only available in OpenCL mode
+  // and follows the OpenCL rules.
```

Comment:
Not salient for this patch, but do we prevent the user from being in OpenCL and SYCL modes at the same time, or do we have to worry about that situation?

Reply:
>> do we prevent the user from being in OpenCL and SYCL modes at the same time, 

Not sure about this. Probably we do not

Reply:
Heh, that may be worth finding out (we can always fix things up later for it). I seem to recall a hazy recollection that users were allowed to mix CUDA and SYCL or OpenCL and SYCL, but perhaps only in host mode. But I don't think this situation is super common, so waiting to resolve it is fine by me.

Reply:
> Heh, that may be worth finding out (we can always fix things up later for it). I seem to recall a hazy recollection that users were allowed to mix CUDA and SYCL or OpenCL and SYCL, but perhaps only in host mode. But I don't think this situation is super common, so waiting to resolve it is fine by me.

I have created https://github.com/intel/llvm/issues/5700 and will take a look at this.